### PR TITLE
Sync test http server to cluster in make cluster-sync

### PR DESCRIPTION
**Pull request by awels**
- Added init/http docker images creation during sync
- Sync init/http image to cluster registry.
- Start http server service during sync.

Signed-off-by: Alexander Wels <awels@redhat.com>

### CODE REVIEW

**Comment by awels** [comment]
@jeffvance @copejon @davidvossel please review. This PR syncs the http/s3 server into the cluster during a make sync, and then starts it. This will be a pre-requisite for running the e2e test.

**Comment by awels** [comment]
I am going to have to test if the service which is created in the the kube-system namespace (should we do it somewhere else?) is exposed to other namespaces or not, if not we need to solve that for the tests.

**Comment by awels** [comment]
Looks like I messed up some stuff, working on fixing the docker make target.

File: Makefile [hunk 1]
```diff
@@ -111,5 +111,12 @@ functest:
 	./hack/build/functests.sh
 
 functest-image-host: WHAT=tools/cdi-func-test-file-host-init
-functest-image-host:  manifests build
+functest-image-host:  manifests docker
```
Comment (copejon) [inline Makefile]: Why build all project binaries and images from scratch when we build the image server?

Comment (copejon) [reply Makefile]: This also creates redundant work for line L115 `./hack/build/build-docker.sh "tools/cdi-func-test-file-host-init tools/cdi-func-test-file-host-http"`

File: hack/build/config.sh [hunk 2]
```diff
@@ -23,7 +23,7 @@ CONTROLLER_MAIN="cmd/${CONTROLLER}"
 IMPORTER_MAIN="cmd/${IMPORTER}"
 CLONER_MAIN="cmd/${CLONER}"
 
-DOCKER_IMAGES="cmd/${CONTROLLER} cmd/${IMPORTER} cmd/${CLONER}"
+DOCKER_IMAGES="cmd/${CONTROLLER} cmd/${IMPORTER} cmd/${CLONER} tools/cdi-func-test-file-host-init tools/cdi-func-test-file-host-http"
```
Comment (copejon) [inline hack/build/config.sh]: Can you extract the `cdi-func-test-file-host-*` names into separate variables

File: Makefile [hunk 3]
```diff
@@ -111,5 +111,12 @@ functest:
...
 	${DO} ./hack/build/build-cdi-func-test-file-host.sh && ./hack/build/build-docker.sh "tools/cdi-func-test-file-host-init tools/cdi-func-test-file-host-http"
+
+cluster-sync-functest-image-init: WHAT = tools/cdi-func-test-file-host-init
+cluster-sync-functest-image-init: functest-image-host
```
Comment (copejon) [inline Makefile]: This isn't syncing anything to the cluster from what I can see.  The prerequisite should probably be `cluster-sync`.

File: Makefile [hunk 4]
```diff
@@ -111,5 +111,12 @@ functest:
...
+
+cluster-sync-functest-image-http: WHAT = tools/cdi-func-test-file-host-http
+cluster-sync-functest-image-http: functest-image-host
```
Comment (copejon) [inline Makefile]: ditto

File: hack/build/config.sh (hunk 2 above)
Comment (awels) [reply hack/build/config.sh]: yes

File: Makefile (hunk 3 above)
Comment (awels) [reply Makefile]: Yeah, I was a little optimistic here, I had run some prerequisite manually and didn't do a make clean to make sure everything got build properly. I will push an update soon that should be much better.

File: cluster/sync.sh [hunk 5]
```diff
@@ -27,4 +27,7 @@ for i in $(seq 1 ${KUBEVIRT_NUM_NODES}); do
     ./cluster/cli.sh ssh "node$(printf "%02d" ${i})" "echo \"${container}\" | xargs \-\-max-args=1 sudo docker pull"
 done
 
+# Install CDI
 ./cluster/kubectl.sh apply -f ./manifests/generated/cdi-controller.yaml
+# Start functional test HTTP server.
+./cluster/kubectl.sh apply -f ./manifests/generated/file-host.yaml
```
[the remark this answers is not in the document]
Comment (awels) [reply cluster/sync.sh]: You make a really good point, but it should be fairly easy to remedy, simply do a delete, and wait for the pods to disappear, then do the create. Or any number of other options to recreate the pods. I think the delete and wait is the simplest though.

[the remark this answers is not in the document]
Comment (davidvossel) [reply cluster/sync.sh]: take a look at how kubevirt cleans the cluster. They use the kubevirt.io take to remove all objects of a specific type. We already have the cdi.kubevirt.io label and can do the same thing. 

https://github.com/kubevirt/kubevirt/blob/master/cluster/clean.sh

[the remark this answers is not in the document]
Comment (awels) [reply cluster/sync.sh]: Nice, makes sense now, cluster-clean calls the cleaner, then in cluster-sync we call clean, and then the actual sync.

[the remark this answers is not in the document]
Comment (awels) [reply cluster/sync.sh]: @davidvossel stole the cleanup from kubevirt. We are now actually syncing.

[the remark this answers is not in the document]
Comment (davidvossel) [reply cluster/sync.sh]: @awels yep, looks good!

**Comment by awels** [comment]
@copejon I believe the latest version addresses your concerns, please re-review.

**Comment by davidvossel** [comment]
I tested this locally and it worked for me

**Comment by jeffvance** [comment]
lgtm
@copejon you have a review approval request.

**Pull request merged**